composables/cache.ts

```diff
@@ -57,27 +45,16 @@ export function fetchAccountById(id?: string | null): Promise<mastodon.v1.Accoun
   if (cached)
     return Promise.resolve(cached)
 
-  let fetchPromise = fetchPromises.get(key)
-  if (!fetchPromise) {
-    const domain = getInstanceDomainFromServer(server)
-    fetchPromise = useMastoClient().v1.accounts.$select(id).fetch()
-      .then((r) => {
-        if (r.acct && !r.acct.includes('@') && domain)
-          r.acct = `${r.acct}@${domain}`
-
-        cacheAccount(r, server, true)
-        return Promise.resolve(r)
-      })
-      .finally(() => fetchPromises.delete(key))
-    fetchPromises.set(key, fetchPromise)
-  }
-
-  const promise = new Promise<any>((resolve) => {
-    fetchPromise!.then(resolve)
-  })
-  if (!cache.has(key))
-    cache.set(key, promise)
-
+  const domain = getInstanceDomainFromServer(server)
+  const promise = useMastoClient().v1.accounts.$select(id).fetch()
+    .then((r) => {
+      if (r.acct && !r.acct.includes('@') && domain)
+        r.acct = `${r.acct}@${domain}`
+
+      cacheAccount(r, server, true)
```

Comment:
Same here, but this one seems to be doing more. It may be good to call `cache.set(key, r)` here and then do whatever else `cacheAccount` does in a separate function. We could do that in another PR though.

Reply:
This pr is about to fix the return on these 3 methods,  should return a promise to avoid errors like `.then not found`

Reply:
I can send a pr tmr to includeyour suggestions

Reply:
Ah, it seems I was looking at a previous commit